File: src/main/scala/com/celestia/factories/GameObjectFactory.scala

package com.celestia.factories

import com.celestia.interfaces._
import com.celestia.models.{Camera, GameWorld, GameState, Triangle}
import com.celestia.strategies.{RenderStrategy, UpdateStrategy}
import graphicslib3D.Matrix3D


/**
 * Singleton Factory class for creating new Game
 * objects.
 */
object GameObjectFactory extends IGameObjectFactory {

  /**
   * Create a new Triangle object
   * @return
   */
  override def makeTriangle:Triangle = {
    new Triangle(
      new Matrix3D(),
      new Matrix3D(),
      new Matrix3D(),
      new UpdateStrategy(),
      new RenderStrategy()
    )
  }

  /**
   * Create  new Game State object
   * @return
   */
  override def initGameState:IGameState={
    new GameState(
      initGameWorld,
      initCamera
    )
  }


  /**
   * Create a new Game World collection
   * @return
   */
  override def initGameWorld:IGameWorld=new GameWorld(List[IGameObject]())


  /**
   * Create a new Camera object
   * @return
   */
  override def initCamera:ICamera=new Camera(
    0,
    0,
    0
  )
//  override def initCamera:ICamera=new Camera(
//    new Matrix3D(),
//    new Matrix3D(),
//    new Matrix3D()
//  )
}

File: src/main/java/com/celestia/factories/GameObjectFactory.java

package com.celestia.factories; 

public class GameObjectFactory implements IGameObjectFactory {
    
    @Override 
    public Triangle makeTriangle() {
        return new Triangle();
    }
    
    @Override 
    public IGameState initGameState() {
        return new GameState(
            initCamera(),
            initGameWorld()
        );
    }
    
    @Override 
    public IGameWorld initGameWorld() {
        return new GameWorld();
    }
    
    @Override 
    public ICamera initCamera() {
        return new Camera();
    }
}
